tests/mocks/ui_mock.py

```diff
@@ -0,0 +1,9 @@
+import sys
+import types
+
+module_name = 'scripts.engine.ui.manager'
+mock_module_name = f'{module_name}_mock'
+mock_module = types.ModuleType(mock_module_name)
+
+sys.modules['scripts.engine.ui.manager'] = mock_module
+sys.modules['scripts.engine.ui.manager'].ui = {}
```

Comment:
This class is for mocking the UI class of the game. If we didn't had this every time we ran a test the whole environment would have to be set up including the pygame window and all the graphical resources. This way we set the module to be a dummy object that does nothing and we can override methods to give use specific results. An example below.